### DISCUSSION TITLE
Frigate 0.8.1 A working example with raspberry Pi camera (IMX219)

### DISCUSSION THREAD

**Discussion by lunard** [blakeblackshear/frigate#678] [Show and tell]
Hello to all, I really need your help to understand how to correctly stream and detect object from a raspberry Pi camera with Frigate 0.8.1.

I installed the camera module on my Raspberry Pi 3A+ and tested 2 way to stream using the rtsp protocol:

- v4l2rtspserver  -W 1920 -H 1080 -F 30 -P 8555 /dev/video0   (path: rtsp://<IP>:8555/unicast)
- raspivid -o - -t 0 -n -w 600 -h 400 -fps 15  | cvlc -vvv stream:///dev/stdin --sout '#rtp{sdp=rtsp://:8554/}' :demux=h264 --h264-fps=15  (path rtsp://<IP>:8554/)

With both methods I'm able to see the stream using VLC
But the 2 methods are not working with Frigate 0.8.1

**For the FIRST one I get these errors:**
ffmpeg.back.detect             ERROR   : [AVHWDeviceContext @ 0x558970671500] libva: /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so init failed
frigate    | ffmpeg.back.detect             ERROR   : [AVHWDeviceContext @ 0x558970671500] Failed to initialise VAAPI connection: -1 (unknown libva error).
frigate    | ffmpeg.back.detect             ERROR   : Device creation failed: -5.
frigate    | ffmpeg.back.detect             ERROR   : [mjpeg @ 0x558970675d00] No device available for decoder: device type vaapi needed for codec mjpeg.
frigate    | ffmpeg.back.detect             ERROR   : Device setup failed for decoder on input stream #0:0 : Input/output error
frigate    | frigate.video                  INFO    : back: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
frigate    | frigate.video                  INFO    : back: ffmpeg process is not running. exiting capture thread...

**For the SECOND one I get these errors:**
frigate    | ffmpeg.back.detect             ERROR   : [rtsp @ 0x558fb119b880] method SETUP failed: 461 Client error
frigate    | ffmpeg.back.detect             ERROR   : rtsp://argo:8554/: Protocol not supported
frigate    | frigate.video                  INFO    : back: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
frigate    | frigate.video                  INFO    : back: ffmpeg process is not running. exiting capture thread...

My configuration is really simple:

 
cameras:
  back:
    ffmpeg:
      inputs:
        - path: rtsp://argo:8554/ 
        #- path: rtsp://argo:8555/unicast 
          roles:
            - detect
    width: 1920
    height: 1080
    fps: 5

ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    
    
Someone could help me to understand how correctly configure stream from the raspi camera and configure Frigate?

Thanks !

**Comment by blakeblackshear** [comment]
1. hardware acceleration needs to be removed since this is a mjpeg stream
2. it doesnt look like that stream is compatible with ffmpeg for some reason

Ideally, you want to find a way to send raw yuv420p data directly to frigate. That way you can avoid the overhead of encoding the video only to decode it again right away.

**Reply by lunard** [reply]
I will try to search the way ;-( thanks !

**Comment by mzac** [comment]
I got it working finally by using v4l2rtspserver on my Pi Zero.

Follow the example here:

https://kevinsaye.wordpress.com/2018/10/17/making-a-rtsp-server-out-of-a-raspberry-pi-in-15-minutes-or-less/

**Reply by lunard** [reply]
Hello! As described in the post, I also use v4l2rtspserver. It works without problem with VLC, but not with Frigate.
Were you able to use v4l2rtspserver with Frigate ?

**Reply by jshep321** [reply]
Yes, worked for me. cameras section of frigate.yml below.  Make sure you can view the stream on VLC first, as I had a bad picam cable!
```
cameras:
  rtspcam1:
    ffmpeg:
      inputs:
        - path: "rtsp://10.0.0.71:8554/unicast"
          roles:
            - detect
            - rtmp
```
